src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -165,16 +165,16 @@ private static async ValueTask<(bool, T?)> TryResolveBodyAsync<T>(HttpContext ht
             return T.BindAsync(context, parameter);
         }
 
-        private static ValueTask<(bool, T?)> TryResolveJsonBodyOrServiceAsync<T>(HttpContext httpContext, bool isOptional, IServiceProviderIsService? serviceProviderIsService = null)
+        private static Func<HttpContext, bool, ValueTask<(bool, T?)>> ResolveJsonBodyOrService<T>(IServiceProviderIsService? serviceProviderIsService = null)
         {
             if (serviceProviderIsService is not null)
             {
                 if (serviceProviderIsService.IsService(typeof(T)))
                 {
-                    return new ValueTask<(bool, T?)>((true, httpContext.RequestServices.GetService<T>()));
+                    return (httpContext, isOptional) => new ValueTask<(bool, T?)>((true, httpContext.RequestServices.GetService<T>()));
                 }
             }
-            return TryResolveBodyAsync<T>(httpContext, isOptional);
+            return (httpContext, isOptional) => TryResolveBodyAsync<T>(httpContext, isOptional);
```

Comment:
```suggestion
            return static (httpContext, isOptional) => TryResolveBodyAsync<T>(httpContext, isOptional);
```

I'm assuming both of these lambdas can be static of course.